psi4/driver/driver_findif.py

```diff
@@ -0,0 +1,214 @@
+                       "\tGenerating geometries for use with {:d}-point formula.\n"
+                       "\tDisplacement size will be {:6.2e}.\n".format(print_msg, num_pts, disp_size))
+
+    # TODO: Replace with a generator from a stencil to a set of points.
+    pts_dict = {
+        3: {
+            "sym": ((-1, ), (1, )),
+            "asym": ((-1, ), ),
+            "off": ((1, 1), (-1, -1))
+        },
+        5: {
+            "sym": ((-2, ), (-1, ), (1, ), (2, )),
+            "asym": ((-2, ), (-1, )),
+            "off": ((-1, -2), (-2, -1), (-1, -1), (1, -1), (-1, 1), (1, 1), (2, 1), (1, 2))
+        }
+    }
+
+    # We won't need this variable for a while, but let's validate num_pts early.
+    try:
+        disps = pts_dict[num_pts]
+    except KeyError:
+        raise ValidationError("FINDIF: Invalid number of points!")
+
+    # Get settings for CdSalcList, then get it.
+    method_allowed_irreps = 0x1 if mode == "1_0" else 0xFF
+    t_project = not core.get_global_option("EXTERN") and not core.get_global_option("PERTURB_H")
+    # core.get_local_option returns an int, but CdSalcList expect a bool... Re-cast as bool.
+    r_project = t_project and bool(core.get_local_option("FINDIF", "FD_PROJECT"))
+    salc_list = core.CdSalcList(mol, method_allowed_irreps, t_project, r_project)
+
+    # Convention: A capital N at the start of variable means "number of"
+    Natom = mol.natom()
+    Nirrep = salc_list.nirrep()
+    Nsalc = salc_list.ncd()
+
+    if print_lvl:
+        core.print_out("\tNumber of atoms is {:d}.\n".format(Natom))
+        if method_allowed_irreps != 0x1:
+            core.print_out("\tNumber of irreps is {:d}.\n".format(Nirrep))
+        core.print_out("\tNumber of {!s}SALCs is {:d}.\n".format("" if method_allowed_irreps != 0x1 else "symmetric ",
+                                                                 Nsalc))
+        core.print_out("\tTranslations projected? {:d}. Rotations projected? {:d}.\n".format(t_project, r_project))
+
+    # Convention: x_pi means x_per_irrep. The ith element is x for irrep i, with Cotton ordering.
+    # Here, x is a list of indices of CdSALCs belonging to the specified irrep.
+    salc_indices_pi = [[] for irrep in range(Nirrep)]
+
+    # Validate that we have an irrep matching the user-specified irrep, if any.
+    try:
+        salc_indices_pi[freq_irrep_only]
+    except (TypeError, IndexError):
+        if freq_irrep_only != -1:
+            raise ValidationError("FINDIF: Irrep value not in valid range.")
+
+    # Populate salc_indices_pi for all irreps.
+    for i, salc in enumerate(salc_list):
+        salc_indices_pi[salc.irrep_index()].append(i)
+
+    # If the method allows more than one irrep, print how the irreps partition the SALCS.
+    if print_lvl and method_allowed_irreps != 0x1:
+        core.print_out("\tIndex of SALCs per irrep:\n")
+        for i in range(Nirrep):
+            if print_lvl > 1 or freq_irrep_only in {i, -1}:
```

Comment:
Thanks, I had to look that up to see that curly braces are an alternate way to initialize `set()`.